corehq/apps/app_manager/tests/test_tasks.py

```diff
@@ -88,3 +95,28 @@ def test_dependencies_feature_metrics_not_triggerd(self):
         with patch("corehq.apps.app_manager.tasks.metrics_counter") as metric_counter_mock:
             make_app_build(factory.app, "comment", user_id="user_id")
             metric_counter_mock.assert_not_called()
+
+    def test_refresh_data_dictionary_from_app(self):
+        factory = AppFactory(build_version='2.56.0')
+        m0, f0 = factory.new_basic_module('update', 'case')
+        factory.form_requires_case(f0, update={'texture': '/data/texture'})
+        f0.source = get_simple_form(xmlns=f0.unique_id)
+        factory.form_uses_usercase(f0, update={
+            'favorite_color': ConditionalCaseUpdate(question_path='/data/favorite_color')
+        })
+        m1, f1 = factory.new_advanced_module('advanced', 'case')
+        factory.form_requires_case(f1, 'person')
+        factory.app.save()
+        self.addCleanup(factory.app.delete)
+
+        with patch('corehq.apps.app_manager.tasks.get_app') as get_app:
+            get_app.return_value = factory.app
+            refresh_data_dictionary_from_app(self.domain, factory.app.get_id)
```

Comment:
[`CELERY_TASK_ALWAYS_EAGER` is true for tests](https://github.com/dimagi/commcare-hq/blob/f1f542a2f7e2103bc07b5913a2cdba18649dd951/testsettings.py#L72). Does this not get called automatically when the app is saved a few lines above?